Sources/LanguageServerProtocolJSONRPC/JSONRPCConnection.swift

```diff
@@ -293,27 +253,19 @@ public final class JSONRPCConnection: Connection {
       state = .running
       self.receiveHandler = receiveHandler
       self.closeHandler = closeHandler
+    }
 
-      receiveIO.read(offset: 0, length: Int.max, queue: queue) { done, data, errorCode in
-        guard errorCode == 0 else {
-          #if !os(Windows)
-          if errorCode != POSIXError.ECANCELED.rawValue {
-            logger.fault("IO error reading \(errorCode)")
+    self.inFD.readabilityHandler = { fileHandle in
+      let data = fileHandle.availableData
```

Comment:
I unfortunately can’t remember the example when `fileHandle.availableData` has issues but I’ve been bitten by it more than once. I just remember that it’s a very delicate scenario where it misbehaves and it’s a pain to debug when it happens. So, I really want to use `read(upToCount: Int.max))` here.